pkg/controller/messaginginfra/broker/controller.go

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package broker
+
+import (
+	"context"
+	"fmt"
+
+	logr "github.com/go-logr/logr"
+
+	v1beta2 "github.com/enmasseproject/enmasse/pkg/apis/enmasse/v1beta2"
+	"github.com/enmasseproject/enmasse/pkg/controller/messaginginfra/cert"
+	"github.com/enmasseproject/enmasse/pkg/controller/messaginginfra/common"
+	"github.com/enmasseproject/enmasse/pkg/util"
+	"github.com/enmasseproject/enmasse/pkg/util/install"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	resource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type BrokerController struct {
+	client         client.Client
+	scheme         *runtime.Scheme
+	certController *cert.CertController
+}
+
+func NewBrokerController(client client.Client, scheme *runtime.Scheme, certController *cert.CertController) *BrokerController {
+	return &BrokerController{
+		client:         client,
+		scheme:         scheme,
+		certController: certController,
+	}
+}
+
+func getBrokerLabels(infra *v1beta2.MessagingInfra) map[string]string {
+	labels := make(map[string]string, 0)
+	labels["component"] = fmt.Sprintf("broker-%s", infra.Name)
```

Comment:
Wouldn't it make sense to have the infra name also in an dedicated label?

Reply:
Agreed, will add that.